src/viam/sdk/common/linear_algebra.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+#include <boost/qvm/vec.hpp>
+#include <boost/qvm/vec_traits.hpp>
+
+#include <viam/api/common/v1/common.pb.h>
+namespace viam {
+namespace sdk {
+
+// In the future, we may wish to inline this whole class
+// for performance reasons.
+
+class Vector3 {
+   public:
+    using scalar_type = double;
+
+    Vector3(scalar_type x, scalar_type y, scalar_type z) : data_{x, y, z} {};
```

Comment:
If this class is all out-of-line for now then these constructors probably should be too.